consensus/types.go

```diff
@@ -210,71 +221,68 @@ type SiafundOutput struct {
-// UnlockConditions to be revealed without revealing all of the elements. The
-// PublicKeys are put in the middle because Timelock and NumSignatures are each
-// low entropy fields. They can be protected by having random public keys next
-// to them. (The use case for revealing public keys but not timestamps or
-// numsignatures is unknown, but it is available nonetheless).
+// If 'NumSignatures' == 0, the UnlockConditions are effectively "anyone can
+// unlock." If 'NumSignatures' > len('PublicKeys'), then the UnlockConditions
+// cannot be fulfilled under any circumstances.
 type UnlockConditions struct {
 	Timelock      BlockHeight
 	PublicKeys    []SiaPublicKey
 	NumSignatures uint64
 }
 
-// A SiaPublicKey is a public key prefixed by a specifier. The specifier
-// indicates the algorithm used for signing and verification, and the byte
-// slice contains the actual public key. Unrecognized algorithms will always
-// verify, which allows new algorithms to be soft forked into the protocol.
-// This is useful for safeguarding against the cryptographic break of an
-// algorithm in use, and useful for taking advantage of any new algorithms that
-// get discovered.
+// A SiaPublicKey is a public key prefixed by a Specifier. The Specifier
+// indicates the algorithm used for signing and verification. Unrecognized
+// algorithms will always verify, which allows new algorithms to be added to
+// the protocol via a soft-fork.
 type SiaPublicKey struct {
 	Algorithm Specifier
-	Key       []byte
-}
-
-// A TransactionSignature signs a single PublicKey in one of the
-// UnlockConditions of the transaction. Which UnlockConditions is indicated by
-// the ParentID, which is the same as the crypto.Hash representation of the
-// ParentID of the element that presented the UnlockConditions. The
-// PublicKeyIndex indicates which public key of the UnlockConditions is doing
-// the signing. The Timelock prevents the TransactionSignature from being used
-// until a certain block height. CoveredFields indicates which parts of the
-// transaction have been signed, and the Signature contains the actual
-// signature.
+	Key       string
+}
+
+// A TransactionSignature is a signature that is included in the transaction.
+// The signature should correspond to a public key in one of the
+// UnlockConditions of the transaction. This key is specified first by
+// 'ParentID', which specifies the UnlockConditions, and then
+// 'PublicKeyIndex', which indicates the key in the UnlockConditions. There
+// are three types that use UnlockConditions: SiacoinInputs, SiafundInputs,
+// and FileContractTerminations. Each of these types also references a
+// ParentID, and this is the hash that 'ParentID' must match. The 'Timelock'
+// prevents the signature from being used until a certain height.
+// 'CoveredFields' indicates which parts of the transaction are being signed;
+// see CoveredFields.
+//
+// TODO: use an enum here?
```

Comment:
probably not use an enum, since this is a type that's going over the wire and needs to be clearly defined.